src/main/java/org/prebid/server/auction/VideoRequestFactory.java

```diff
@@ -85,7 +87,22 @@ private BidRequestVideo parseRequest(RoutingContext context) {
         }
 
         try {
-            return mapper.decodeValue(body, BidRequestVideo.class);
+            BidRequestVideo bidRequestVideo = mapper.decodeValue(body, BidRequestVideo.class);
+            if (bidRequestVideo.getDevice() == null) {
+                bidRequestVideo = bidRequestVideo.toBuilder().device(Device.builder().build()).build();
+            }
+            if (StringUtils.isEmpty(bidRequestVideo.getDevice().getUa())) {
+                final String userAgentHeader = context.request().getHeader(HttpUtil.USER_AGENT_HEADER);
+                if (StringUtils.isEmpty(userAgentHeader)) {
+                    throw new InvalidRequestException("Device.UA and User-Agent Header is not presented");
+                }
+                bidRequestVideo = bidRequestVideo.toBuilder()
+                        .device(bidRequestVideo.getDevice().toBuilder()
+                                .ua(userAgentHeader)
+                                .build())
+                        .build();
+            }
```

Comment:
First of all. Don't you thing that 
1 Check if Device is null 
2 Create new Device if it is Null and insert into request. 
3 call `bidRequestVideo.getDevice()` 4 times 
4 ...
5 Then create new request. And create new device again. And set UA.